torchbenchmark/models/DALLE2_pytorch/__init__.py

```diff
@@ -0,0 +1,132 @@
+import argparse
+import random
+import torch
+
+import numpy as np
+
+from torchbenchmark.util.env_check import set_random_seed
+
+from dalle2_pytorch import DALLE2, Unet, Decoder, DiffusionPriorNetwork, DiffusionPrior, OpenAIClipAdapter
+import typing
+
+torch.backends.cudnn.deterministic = False
+torch.backends.cudnn.benchmark = False
+from pathlib import Path
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import COMPUTER_VISION
+
+import io
+
+
+class Model(BenchmarkModel):
+    task = COMPUTER_VISION.GENERATION
+    DEFAULT_TRAIN_BSIZE = 16
```

Comment:
Train batch size can be critical to correctness. Can we have a comment reference of the original batch size used in the origin repo?

Reply:
`loss = diffusion_prior_trainer(text, images, max_batch_size = 4)
diffusion_prior_trainer.update()  # this will update the optimizer as well as the exponential moving averaged diffusion prior

# after much of the above three lines in a loop
# you can sample from the exponential moving average of the diffusion prior identically to how you do so for DiffusionPrior

image_embeds = diffusion_prior_trainer.sample(text, max_batch_size = 4) # (512, 512) - exponential moving averaged image embeddings`

Grabbed from DALLE2-pytorch, maybe DEFAULT_TRAIN_BSIZE = DEFAULT_EVAL_BSIZE = 4?

Reply:
Yes, batch size =4 looks good to me. We can't change the batch size for train, but for eval, we can increase/decrease it on demand.
- If the gpu is too idle, we should increase the batch size to make sure gpu is well utilized
- If the test time is too long > 1000 ms, we should decrease the batch size to reduce test time.